pages/api/v1/listing/get-similar-posts.ts

```diff
@@ -0,0 +1,256 @@
+				}
+			});
+			posts = await Promise.all(postsPromise);
+		}
+	}
+
+	let result: any = [];
+	if (subsquidRes['data'].proposal && posts) {
+		result = subsquidData
+			.map((proposal: any) => posts.filter((post: any) => proposal.index === post?.post_id))
+			.flat()
+			.map((match: any) => {
+				// console.log(`ids from active tags -> ${match}`);
+				return match;
+			});
+	}
+
+	if (result.length < 3) {
+		result = subsquidData
+			.filter((proposal: any) => proposal.trackNumber === trackNumber && proposal.trackNumber !== null)
+			.map((proposal: any) => {
+				// console.log(`ids from active tracks -> ${proposal.index}`);
+				return proposal.index;
+			});
+	}
+
+	// for categories (start)
+	let postTopics: any;
+	if (result.length > 3) {
+		onChainCollRef = postsByTypeRef(network, strProposalType as ProposalType);
+		const postSnapshottopic = await onChainCollRef.where('topic.id', '==', topicId).get();
+		const postsTopicPromise = postSnapshottopic.docs.map(async (doc: any) => {
+			if (doc && doc.exists) {
+				const docData = doc.data();
+				if (docData) {
+					let subsquareTitle = '';
+					if (docData?.title === '' || docData?.title === undefined) {
+						const res = await getSubSquareContentAndTitle(strProposalType, network, docData.id);
+						subsquareTitle = res?.title;
+					}
+					const created_at = docData.created_at;
+					const { topic, topic_id } = docData;
+
+					return {
+						created_at: created_at?.toDate ? created_at?.toDate() : created_at,
+						gov_type: docData?.gov_type,
+						isSpam: docData?.isSpam || false,
+						isSpamReportInvalid: docData?.isSpamReportInvalid || false,
+						post_id: docData.id,
+						proposer: getProposerAddressFromFirestorePostData(docData, network),
+						spam_users_count:
+							docData?.isSpam && !docData?.isSpamReportInvalid ? Number(process.env.REPORTS_THRESHOLD || 50) : docData?.isSpamReportInvalid ? 0 : docData?.spam_users_count || 0,
+						tags: docData?.tags || [],
+						title: docData?.title || subsquareTitle || null,
+						topic: topic
+							? topic
+							: isTopicIdValid(topic_id)
+							? {
+									id: topic_id,
+									name: getTopicNameFromTopicId(topic_id)
+							  }
+							: getTopicFromType(strProposalType as ProposalType),
+						user_id: docData?.user_id || 1,
+						username: docData?.username
+					};
+				}
+			}
+		});
+		postTopics = await Promise.all(postsTopicPromise);
+		if (subsquidRes['data'].proposal && postTopics) {
+			result = subsquidData
+				.map((proposal: any) => postTopics.filter((postTopic: any) => proposal.index === postTopic?.post_id))
+				.flat()
+				.map((match: any) => {
+					// console.log(`ids from active subtracks -> ${match}`);
+					return match;
+				});
+		}
+	}
+
+	if (result.length < 3) {
+		result = subsquidData.map((proposal: any) => {
+			// console.log(`ids from non matches -> ${proposal.index}`);
+			return proposal.index;
+		});
+	}
+	result = result.filter((number: any) => number !== postId);
```

Comment:
Avoid this loop. Don't fetch the current referenda from firestore and subsquid

Reply:
sir without this, joh proposal open hai uski id bhi similar proposal ki ids wli array m store ho jyegi

Reply:
use not_eq: postId in queries from subsquid/firestore to exclude current proposal